package/lib/src/modal/modal_sheet.dart

```diff
@@ -47,6 +48,8 @@ class ModalSheetPage<T> extends Page<T> {
 
   final Curve transitionCurve;
 
+  final Widget? barrier;
```

Comment:
I think this should be a builder rather than a widget because the barrier is lazily built in `buildModalBarrier`. The signature could be something like `final ModalSheetBarrierBuilder? barrierBuilder` where `ModalSheetBarrierBuilder` is:
```dart
typedef ModalSheetBarrierBuilder = Widget Function(PageRoute route, VoidCallback onDismiss);
```